JuliaStream.jl/README.md

```diff
@@ -0,0 +1,30 @@
+JuliaStream.jl
+==============
+
+This is an implementation of BabelStream in Julia which contains the following variants:
+
+ * `PlainStream.jl` - Single threaded `for`
+ * `ThreadedStream.jl` - Threaded implementation with `Threads.@threads` macros
+ * `DistributedStream.jl` - Process based parallelism with `@distributed` macros
+ * `CUDAStream.jl` - Direct port of BabelStream's native CUDA implementation using [CUDA.jl](https://github.com/JuliaGPU/CUDA.jl)
+ * `AMDGPUStream.jl` - Direct port of BabelStream's native HIP implementation using [AMDGPU.jl](https://github.com/JuliaGPU/AMDGPU.jl)
+
+### Build & Run
+
+Prerequisites
+
+ * Julia 1.6+
+
+With Julia on path, run the benchmark with:
+
+```shell
+> cd JuliaStream.jl
+> julia --project -e 'import Pkg; Pkg.instantiate()' # only required on first run
+> julia --project src/<IMPL>Stream.jl
+```
+
+**Important:**
+ * Julia is 1-indexed, so N > 1 in `--device N`
+ * Thread count for `ThreadedStream` must be set via the `JULIA_NUM_THREADS` environment variable (e.g `export JULIA_NUM_THREADS=$(nproc)`) otherwise it defaults to 1
+ * `DistributedStream` uses `addprocs()` call directly which defaults to `$(nproc)`, **do not use the `-p <N>` flag** as per the [documentation](https://docs.julialang.org/en/v1/manual/distributed-computing).
```

Comment:
What is the reason behind this? Why don't we want to follow the documented approach of using the `-p` flag?

Reply:
Adding `-p` forks the process before the project environment gets initialised.
The alternative is to setup the environment for each worker process, this lengthens the already slow startup. 

That said, the distributed version is already quite slow so it's probably fine. 
It's less flexible with `addprocs()` anyway as I'll have to propagate the worker count myself from the program args which isn't implemented.

Reply:
Done